sensitivity-computation/src/main/java/com/farao_community/farao/sensitivity_computation/SystematicSensitivityAnalysisInterface.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    public SystematicSensitivityAnalysisInterface(SensitivityComputationParameters defaultParameters, SensitivityComputationParameters fallbackParameters) {
+        this.defaultParameters = defaultParameters;
+        this.fallbackParameters = fallbackParameters;
+    }
+
+    public SystematicSensitivityAnalysisInterface() {
+        this.defaultParameters = new SensitivityComputationParameters();
+    }
+
+    public void setSensitivityProvider(SensitivityProvider sensitivityProvider) {
+        this.sensitivityProvider = sensitivityProvider;
+    }
+
+    public boolean isFallback() {
+        return fallbackMode;
+    }
+
+    /**
+     * Run the systematic sensitivity analysis on the given network and crac, and associates the
+     * SystematicSensitivityAnalysisResult to the given network variant.
+     *
+     * Throw a SensitivityComputationException if the computation fails.
+     */
+    public SystematicSensitivityAnalysisResult run(Network network, Crac crac, Unit defaultUnit) {
+        SensitivityComputationParameters sensitivityComputationParameters = fallbackMode ? fallbackParameters : defaultParameters;
+
+        try {
+            return runWithConfig(network, crac, sensitivityComputationParameters, defaultUnit);
+        } catch (SensitivityComputationException e) {
+            if (!fallbackMode && fallbackParameters != null) { // default mode fails, retry in fallback mode
+                LOGGER.warn("Error while running the sensitivity computation with default parameters, fallback sensitivity parameters are now used.");
+                fallbackMode = true;
+                return run(network, crac, defaultUnit);
+            } else if (!fallbackMode) { // no fallback mode available, throw an exception
+                throw new SensitivityComputationException("Sensitivity computation failed with default parameters. No fallback parameters available.", e);
+            } else { // fallback mode fails, throw an exception
+                throw new SensitivityComputationException("Sensitivity computation failed with all available sensitivity parameters.", e);
+            }
+        }
+    }
+
+    // Method for tests
+    SystematicSensitivityAnalysisResult run(Network network, Crac crac) {
+        return run(network, crac, Unit.AMPERE);
+    }
+
+    /**
+     * Run the systematic sensitivity analysis with given SensitivityComputationParameters, throw a
+     * SensitivityComputationException is the computation fails.
+     */
+    private SystematicSensitivityAnalysisResult runWithConfig(Network network, Crac crac, SensitivityComputationParameters sensitivityComputationParameters, Unit defaultUnit) {
+
+        try {
+            SystematicSensitivityAnalysisResult tempSystematicSensitivityAnalysisResult = SystematicSensitivityAnalysisService
+                .runSensitivity(network, network.getVariantManager().getWorkingVariantId(), sensitivityProvider, sensitivityComputationParameters);
```

Comment:
At least you can add a check on sensitivtyProvider

Reply:
done